great_expectations/core/util.py

```diff
@@ -261,3 +263,39 @@ def ensure_json_serializable(data):
 
 def requires_lossy_conversion(d):
     return d - Context(prec=sys.float_info.dig).create_decimal(d) != 0
+
+
+def substitute_all_strftime_format_strings(
+    data: Union[dict, list, str, Any], datetime_obj: Optional[datetime] = None
+) -> Union[str, Any]:
+    """
+    This utility function will iterate over input data and for all strings, replace any strftime format
+    elements using either the provided datetime_obj or the current datetime
+    """
+
+    datetime_obj: datetime = datetime_obj or datetime.now()
+    if isinstance(data, dict) or isinstance(data, OrderedDict):
+        return {
+            k: substitute_all_strftime_format_strings(v, datetime_obj=datetime_obj)
+            for k, v in data.items()
+        }
+    elif isinstance(data, list):
+        return [
+            substitute_all_strftime_format_strings(el, datetime_obj=datetime_obj)
+            for el in data
+        ]
+    elif isinstance(data, str):
+        return get_datetime_string_from_strftime_format(data, datetime_obj=datetime_obj)
+    else:
+        return data
+
+
+def get_datetime_string_from_strftime_format(
```

Comment:
Yikes: We now have `parse_string_to_datetime()` in `great_expectations/datasource/data_connector/sorter/date_time_sorter.py` and `get_datetime_string_from_strftime_format()` in `great_expectations/core/util.py` -- it would be nice to combine -- if appropriate -- and reduce code duplication...

Reply:
I think these are different though, since in `parse_string_to_datetime`, it takes in a `str` that's semantically a date and returns a `datetime.date` object, and in this function, it takes in a strftime format `str` like `%m%A` and optionally a datetime object, and returns a semantic date `str`. But perhaps these should be together rather than in separate places?

Reply:
@roblim I am guessing that "together" might be nicer...  Your call as to when/how.